sessionctx/variable/sysvar.go

```diff
@@ -140,11 +140,11 @@ var defaultSysVars = []*SysVar{
 		}
 		return nil
 	}},
-	{Scope: ScopeSession, Name: TiDBOptProjectionPushDown, Value: BoolToOnOff(config.GetGlobalConfig().Performance.ProjectionPushDown), skipInit: true, Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeSession, Name: TiDBOptProjectionPushDown, Value: BoolToOnOff(config.GetGlobalConfig().Performance.ProjectionPushDown), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
 		s.AllowProjectionPushDown = TiDBOptOn(val)
 		return nil
 	}},
-	{Scope: ScopeSession, Name: TiDBOptAggPushDown, Value: BoolToOnOff(DefOptAggPushDown), Type: TypeBool, skipInit: true, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeSession, Name: TiDBOptAggPushDown, Value: BoolToOnOff(DefOptAggPushDown), Type: TypeBool, skipInit: false, SetSession: func(s *SessionVars, val string) error {
```

Comment:
Same with this one
```suggestion
	{Scope: ScopeSession, Name: TiDBOptAggPushDown, Value: BoolToOnOff(DefOptAggPushDown), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```

Reply:
Since `DefOptAggPushDown` is false, `TiDBOptAggPushDown` doesn't have the problem like #35083 even if `skipInit` is true. Maybe we can just keep it?

Reply:
The only known required use cases for skipInit are:
1. old-style `INSTANCE` scoped variables (which use `SESSION`). There are only 2 of these left: https://github.com/pingcap/tidb/issues/35190
2. when you set the charset, it automatically sets the default collation for that charset.

We can use a workaround to check if it is a user issuing a `Set` statement for (2), so eventually we should be able to remove skipInit completely. The problem is because it exists, everyone keeps using it incorrectly (adding new cases that skipInit).

Reply:
`skipInit` of `TiDBOptAggPushDown` is removed now.